File: mifosng-android/src/main/java/com/mifos/services/data/GroupPayload.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.services.data;

import java.util.List;

/**
 * Created by nellyk on 1/22/2016.
 */
public class GroupPayload {

    private String groupname;
    private int officeId;
    private boolean active;
    private String activationDate;
    private String submittedOnDate;
    private String externalId;
    private int staffId;
    private String name;
    private List<Integer> clientMembers;
    private String dateFormat = "dd MMMM yyyy";
    private String locale;

    public int getStaffId() {
        return staffId;
    }

    public void setStaffId(int staffId) {
        this.staffId = staffId;
    }

    public String getGroupname() {
        return groupname;
    }

    public void setGroupname(String groupname) {
        this.groupname = groupname;
    }

    public String getGroupName() {
        return groupname;
    }

    public void setGroupName(String centername) {
        this.groupname = centername;
    }

    public int getOfficeId() {
        return officeId;
    }

    public void setOfficeId(int officeId) {
        this.officeId = officeId;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public String getActivationDate() {
        return activationDate;
    }

    public void setActivationDate(String activationDate) {
        this.activationDate = activationDate;
    }

    public String setSubmissionDate() {
        return submittedOnDate;
    }

    public void setSubmissionDate(String submittedOnDate) {
        this.submittedOnDate = submittedOnDate;
    }

    public String getExternalId() {
        return externalId;
    }

    public void setExternalId(String externalId) {
        this.externalId = externalId;
    }
}

File: core/database/src/main/java/com/mifos/core/objects/group/GroupPayload.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.objects.group

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.mifos.core.database.MifosDatabase
import com.mifos.core.model.MifosBaseModel
import kotlinx.parcelize.Parcelize

/**
 * Created by nellyk on 1/22/2016.
 */
@Parcelize
@Entity("GroupPayload")
data class GroupPayload(
    @PrimaryKey(autoGenerate = true)
    @Transient
    var id: Int = 0,

    @ColumnInfo("errorMessage")
    @Transient
    var errorMessage: String? = null,

    @ColumnInfo("officeId")
    var officeId: Int = 0,

    @ColumnInfo("active")
    var active: Boolean = false,

    @ColumnInfo("activationDate")
    var activationDate: String? = null,

    @ColumnInfo("submittedOnDate")
    var submittedOnDate: String? = null,

    @ColumnInfo("externalId")
    var externalId: String? = null,

    @ColumnInfo("name")
    var name: String? = null,

    @ColumnInfo("locale")
    var locale: String? = null,

    @ColumnInfo("dateFormat")
    var dateFormat: String? = null,
) : MifosBaseModel(), Parcelable
